rust-src/concordium_base/src/ecvrf/public.rs

```diff
@@ -94,11 +94,16 @@ impl PublicKey {
     fn mangle_scalar_bits_and_multiply_by_basepoint_to_produce_public_key(
         bits: &mut [u8; 32],
     ) -> PublicKey {
-        bits[0] &= 0b_1111_1000;
-        bits[31] &= 0b_0111_1111;
-        bits[31] |= 0b_0100_0000;
+        // It seems like `from_bits` is the only way to construct an unreduced scalar
+        // that can potentially be grater than the field's order.
 
-        let point = &Scalar::from_bits(*bits) * &constants::ED25519_BASEPOINT_TABLE;
+        // TODO: we use deprecated `from_bits` here, this also requires enabling the
+        // `legacy_compatibility` feature for the `ed25519-dalek` dependency. Maybe ther
+        // is a different way of implementing this.
+        #[allow(deprecated)]
+        let scalar = Scalar::from_bits(clamp_integer(*bits));
```

Comment:
So doesn't `from_bits` do nothing here since https://docs.rs/curve25519-dalek/latest/curve25519_dalek/scalar/fn.clamp_integer.html always produces a reduced integer (that is reduced)?

So we could use other construction functions and just unwrap? Commenting why it's safe of course.

Reply:
`clamp_integer()` produces something that could be bigger than the order.
This function is removed, no mangling is required at this point. The scalar is properly computed during the creation of an `ExpandedSecretKey`.